torchx/schedulers/test/ray_scheduler_test.py

```diff
@@ -298,6 +298,18 @@ def test_requirements(self) -> None:
                 job = req.request
                 self.assertEqual(job.requirements, reqs)
 
+        def test_parse_app_id(self) -> None:
+            from itertools import product
+            addrs = ["0.0.0.0", "www.testexample.com", "addr-of-cluster"]
+            ports = ["65535", ]
+            app_ids = ["a-p-p_id", "app-i-d", "app_id", "app:id"]
+            for addr, port, app_id in product(addrs, ports, app_ids):
+                parsed_addr, parsed_app_id = self._scheduler._parse_app_id(
+                    f"{addr}:{port}-{app_id}"
+                )
+                self.assertEqual(parsed_addr, addr)
```

Comment:
I think this is dropping the port -- port should still be included in the address

Would also be good to add an explicit test where there's no port specified and it should fall back to the old behavior for safety reasons

Reply:
My bad, I didn't see the instruction on how to run tests in the document.